node_modules/gh-core/lib/server.js

```diff
@@ -86,7 +86,12 @@ var setUpServer = module.exports.setUpServer = function(port, config) {
 
             // Track the time and response code for the route
             var requestName = 'http.' + req.method + '.' + url;
-            var statsd = StatsdAPI.getInstance(req.ctx.app.id);
+
+            // In situations where a request fails before we've had a chance to bind a Context, we
+            // need to find a way to still get meaningful information to statsd. Use app id -1 as
+            // a way to indicate there was no app information available
+            var appId = (req.ctx && req.ctx.app && req.ctx.app.id) || -1;
```

Comment:
It might be *slightly* better to check for `req.ghApp` rather than `req.ctx.app` as that one gets added on to the requests a few middleware layers earlier.